python/demos/demo_nitsche_meshties.py

```diff
@@ -0,0 +1,207 @@
+    values = np.hstack([top_values, bottom_values, surface_values, sbottom_values])
+    sorted_facets = np.argsort(indices)
+    facet_marker = meshtags(mesh, tdim - 1, indices[sorted_facets], values[sorted_facets])
+
+    # mark the whole domain
+    cells = np.arange(mesh.topology.index_map(tdim).size_local
+                      + mesh.topology.index_map(tdim).num_ghosts, dtype=np.int32)
+    domain_marker = meshtags(mesh, tdim, cells, np.full(cells.shape, 1, dtype=np.int32))
+
+    # Function, TestFunction, TrialFunction and measures
+    V = VectorFunctionSpace(mesh, ("CG", 1))
+    u = Function(V)
+    v = ufl.TestFunction(V)
+    w = ufl.TrialFunction(V)
+    dx = ufl.Measure("dx", domain=mesh, subdomain_data=domain_marker)
+    ds = ufl.Measure("ds", domain=mesh, subdomain_data=facet_marker)
+    h = ufl.CellDiameter(mesh)
+    n = ufl.FacetNormal(mesh)
+
+    # Compute lame parameters
+    E = args.E
+    nu = args.nu
+    mu_func, lambda_func = lame_parameters(False)
+    mu = mu_func(E, nu)
+    lmbda = lambda_func(E, nu)
+    sigma = sigma_func(mu, lmbda)
+
+    # dictionary with problem parameters
+    gamma = args.gamma
+    theta = args.theta
+    problem_parameters = {"mu": mu, "lambda": lmbda, "gamma": E * gamma, "theta": theta}
+
+    J = ufl.inner(sigma(w), epsilon(v)) * dx
+
+    # traction (neumann) boundary condition on mesh boundary with tag 3
+    t = Constant(mesh, ScalarType((0.0, 0.5, 0.0)))
+    F = ufl.inner(t, v) * ds(neumann_bdy)
+
+    # Dirichlet bdry conditions
+    g = Constant(mesh, ScalarType((0.0, 0.0, 0.0)))
+    if args.lifting:
+        bdy_dofs = locate_dofs_topological(V, tdim - 1, facet_marker.find(dirichlet_bdy))  # type: ignore
+        bcs = [dirichletbc(g, bdy_dofs, V)]
+    else:
+        bcs = []
+        J += - ufl.inner(sigma(w) * n, v) * ds(dirichlet_bdy)\
+            - theta * ufl.inner(sigma(v) * n, w) * \
+            ds(dirichlet_bdy) + E * gamma / h * ufl.inner(w, v) * ds(dirichlet_bdy)
+        F += - theta * ufl.inner(sigma(v) * n, g) * \
+            ds(dirichlet_bdy) + E * gamma / h * ufl.inner(g, v) * ds(dirichlet_bdy)
+
+    # body forces
+    f = Constant(mesh, ScalarType((0.0, 0.5, 0.0)))
+    F += ufl.inner(f, v) * dx
+
+    # Solver options
+    ksp_tol = 1e-10
+
+    # for debugging use petsc_options = {"ksp_type": "preonly", "pc_type": "lu"}
+    petsc_options = {
+        "matptap_via": "scalable",
+        "ksp_type": "cg",
+        "ksp_rtol": ksp_tol,
+        "ksp_atol": ksp_tol,
+        "pc_type": "gamg",
+        "pc_mg_levels": 3,
+        "pc_mg_cycles": 1,   # 1 is v, 2 is w
+        "mg_levels_ksp_type": "chebyshev",
+        "mg_levels_pc_type": "jacobi",
+        "pc_gamg_type": "agg",
+        "pc_gamg_coarse_eq_limit": 100,
+        "pc_gamg_agg_nsmooths": 1,
+        "pc_gamg_sym_graph": True,
+        "pc_gamg_threshold": 1e-3,
+        "pc_gamg_square_graph": 2,
+    }
```

Comment:
Maybe we should add an input flag called --debug (true/false) that selects the options so we do not have to go into the code to change the options?